src/plugins/utils/Utils.ts

```diff
@@ -257,32 +261,96 @@ class Utils {
+            let modulePath;
+            if (paths !== undefined) {
+                modulePath = customReq.resolve(name, {paths});
+            } else {
+                modulePath = customReq.resolve(name);
+            }
+
             return parse(modulePath);
         } catch (err) {
             return {};
         }
     }
 
-    static enrichModulePath(paths: string[]) {
-        if (process.env.LAMBDA_TASK_ROOT) {
-            if (paths === undefined) {
-                paths = [process.env.LAMBDA_TASK_ROOT];
-            } else if (paths.indexOf(process.env.LAMBDA_TASK_ROOT) === -1) {
-                paths.push(process.env.LAMBDA_TASK_ROOT);
+    static doInstrument(lib: any, libs: any[], basedir: string, moduleName: string,
+                        version: string, wrapper: any, config?: any): any {
+        let isValid = false;
+        if (version) {
+            const moduleValidator = new ModuleVersionValidator();
+            const isValidVersion = moduleValidator.validateModuleVersion(basedir, version);
+            if (!isValidVersion) {
+                ThundraLogger.getInstance().error(
+                    `Invalid module version for ${moduleName} integration. Supported version is ${version}`);
+            } else {
+                isValid = true;
+            }
+        } else {
+            isValid = true;
+        }
+        if (isValid) {
+            if (!lib[thundraWrapped]) {
+                wrapper(lib, config, moduleName);
+                lib[thundraWrapped] = true;
+                libs.push(lib);
             }
         }
-        return paths || [];
+    }
+
+    static instrument(moduleNames: string[], version: string, wrapper: any,
+                      unwrapper?: any, config?: any, paths?: string[], fileName?: string): any {
+        const libs: any[] = [];
+        const hooks: any[] = [];
+        for (const moduleName of moduleNames) {
+            const requiredLib = Utils.tryRequire(fileName ? path.join(moduleName, fileName) : moduleName, paths);
+            if (requiredLib) {
+                if (version) {
+                    const {basedir} = Utils.getModuleInfo(moduleName);
+                    if (!basedir) {
+                        ThundraLogger.getInstance().error(`Base directory is not found for the package ${moduleName}`);
+                        return;
+                    }
+                    Utils.doInstrument(requiredLib, libs, basedir, moduleName, version, wrapper, config);
+                } else {
+                    Utils.doInstrument(requiredLib, libs, null, moduleName, null, wrapper, config);
+                }
+            }
+            const hook = Hook(moduleName, {internals: true}, (lib: any, name: string, basedir: string) => {
+                if (name === moduleName) {
+                    Utils.doInstrument(lib, libs, basedir, moduleName, version, wrapper, config);
+                }
+                hooks.push(hook);
```

Comment:
I think, this should be outside of Hook block as in all cases (even though instrumentation is not applied during module load) hooks need to be unhook. I mean
```js
            const hook = Hook(moduleName, {internals: true}, (lib: any, name: string, basedir: string) => {
                if (name === moduleName) {
                    Utils.doInstrument(lib, libs, basedir, moduleName, version, wrapper, config);
                }
                return lib;
            });
            hooks.push(hook);
```